### PR TITLE
Add mandatory_tags and exclude_tags for scaleway

### PR DESCRIPTION
**Pull request by Sh4d1**
Signed-off-by: Patrik Cyvoct <patrik@ptrk.io>

##### SUMMARY

Add the possibilty to add mandatory tags and to exclude tags in Scaleway's dynamic inventory

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

lib/ansible/plugins/inventory/scaleway.py

cc @sieben

### CODE REVIEW

File: lib/ansible/plugins/inventory/scaleway.py [hunk 1]
```diff
@@ -213,24 +298,37 @@ def _fill_host_variables(self, host, server_info):
...
+    def match_groups(self, server_info, tags, mandatory_tags, exclude_tags):
         server_zone = extract_zone(server_info=server_info)
         server_tags = extract_tags(server_info=server_info)
 
         # If a server does not have a zone, it means it is archived
         if server_zone is None:
             return set()
+        
+        # return empty set if the server have an excluded tag
+        if exclude_tags is not None and len(set(server_tags).intersection(exclude_tags)) != 0:
```
Comment (remyleone) [inline lib/ansible/plugins/inventory/scaleway.py]: remove len =! 0 and use boolean value of set

File: lib/ansible/plugins/inventory/scaleway.py [hunk 2]
```diff
@@ -213,24 +298,37 @@ def _fill_host_variables(self, host, server_info):
...
+            return set()
+    
+        # if all the mandatory_tags are not present on the server, return empty
+        if mandatory_tags is not None and not set(mandatory_tags).issubset(set(server_tags)):
+            return set()
+        # if we have some mandatory tags and they are ok, we add them to tags if tags is empty
```
Comment (remyleone) [inline lib/ansible/plugins/inventory/scaleway.py]: matching tags

Comment (remyleone) [reply lib/ansible/plugins/inventory/scaleway.py]: Refactor comment

**Review by remyleone — COMMENTED** [review]
Looks good :)

**Comment by Sh4d1** [comment]
@sieben changed!

**Comment by sieben** [comment]
@Sh4d1 Can you fix all the CI bugs?

**Comment by sieben** [comment]
?? What went wrong?

**Comment by Sh4d1** [comment]
@sieben had to do the PR from another branch in my fork

**Comment by sieben** [comment]
Could you put a link to the new PR here?

**Comment by Sh4d1** [comment]
The new PR is opened here : https://github.com/ansible/ansible/pull/53767

**Pull request closed without merging**